tests/test_parser/test_properties/test_model_property.py

```diff
@@ -123,3 +123,77 @@ def test_resolve_references(mocker):
     assert all(not p.required for p in model.optional_properties)
+
+
+def test_resolve_references_nested_allof(mocker):
+    import openapi_python_client.schema as oai
+    from openapi_python_client.parser.properties import build_model_property
+
+    schemas = {
+        "RefA": oai.Schema.construct(
+            title=mocker.MagicMock(),
+            description=mocker.MagicMock(),
+            required=["String"],
+            properties={
+                "String": oai.Schema.construct(type="string"),
+                "Enum": oai.Schema.construct(type="string", enum=["aValue"]),
+                "DateTime": oai.Schema.construct(type="string", format="date-time"),
+            },
+        ),
+        "RefB": oai.Schema.construct(
+            title=mocker.MagicMock(),
+            description=mocker.MagicMock(),
+            required=["DateTime"],
+            properties={
+                "Int": oai.Schema.construct(type="integer"),
+                "DateTime": oai.Schema.construct(type="string", format="date-time"),
+                "Float": oai.Schema.construct(type="number", format="float"),
+            },
+        ),
+        # Intentionally no properties defined
+        "RefC": oai.Schema.construct(
+            title=mocker.MagicMock(),
+            description=mocker.MagicMock(),
+        ),
+    }
+
+    model_schema = oai.Schema.construct(
+        type="object",
+        properties={
+            "Key": oai.Schema.construct(
+                allOf=[
+                    oai.Reference.construct(ref="#/components/schemas/RefA"),
+                    oai.Reference.construct(ref="#/components/schemas/RefB"),
+                    oai.Reference.construct(ref="#/components/schemas/RefC"),
+                    oai.Schema.construct(
+                        title=mocker.MagicMock(),
+                        description=mocker.MagicMock(),
+                        required=["Float"],
+                        properties={
+                            "String": oai.Schema.construct(type="string"),
+                            "Float": oai.Schema.construct(type="number", format="float"),
+                        },
+                    ),
+                ]
+            ),
+        }
+    )
+
+    components = {**schemas, "Model": model_schema}
+
+    from openapi_python_client.parser.properties import Schemas
+
+    schemas_holder = Schemas()
+    model, schemas_holder = build_model_property(
+        data=model_schema, name="Model", required=True, schemas=schemas_holder, parent_name=None
+    )
+    model.resolve_references(components, schemas_holder)
+    assert sorted(p.name for p in model.required_properties) == []
+    assert sorted(p.name for p in model.optional_properties) == ["Key"]
+    assert all(not p.required for p in model.optional_properties)
+
+    key_property = model.optional_properties[0]
+    assert sorted(p.name for p in key_property.required_properties) == ["DateTime", "Float", "String"]
```

Comment:
Currently fails with
```
>       assert sorted(p.name for p in key_property.required_properties) == ["DateTime", "Float", "String"]
E       AssertionError: assert ['Float'] == ['DateTime', ...at', 'String']
E         At index 0 diff: 'Float' != 'DateTime'
E         Right contains 2 more items, first extra item: 'Float'
E         Use -v to get the full diff
```
I can't figure out why.

Reply:
@forest-benchling I think this is because you only resolved references on the parent model, not on the `key_property`.

This should work if you do 
```suggestion
    assert isinstance(key_property, ModelProperty)
    key_property.resolve_references(components, schemas_holder)
    assert sorted(p.name for p in key_property.required_properties) == ["DateTime", "Float", "String"]
```


When you actually run the code generator, all models are resolved in `build_schemas` but we don't use that code path here.

Reply:
wooo thanks @packyg